core/lib/dal/src/tee_proof_generation_dal.rs

```diff
@@ -17,14 +19,23 @@ pub struct TeeProofGenerationDal<'a, 'c> {
     pub(crate) storage: &'a mut Connection<'c, Core>,
 }
 
+#[derive(Debug, Clone, sqlx::FromRow)]
+pub struct L1BatchNumberModel {
+    pub l1_batch_number: i64,
+}
+
 impl TeeProofGenerationDal<'_, '_> {
-    pub async fn get_next_batch_to_be_proven(
+    pub async fn lock_batch_for_proving(
         &mut self,
         tee_type: TeeType,
         processing_timeout: Duration,
+        min_batch_number: Option<L1BatchNumber>,
     ) -> DalResult<Option<L1BatchNumber>> {
         let processing_timeout = pg_interval_from_duration(processing_timeout);
-        let query = sqlx::query!(
+        let min_batch_condition = min_batch_number.map_or(String::new(), |min_batch| {
```

Comment:
AFAIU, it's possible to use [`match_query_as`](https://github.com/matter-labs/zksync-era/blob/87768755e8653e4be5f29945b56fd05a5246d5a8/core/lib/db_connection/src/macro_utils.rs#L84) macro here; you can see some examples of usage in the DAL crate (e.g., [here](https://github.com/matter-labs/zksync-era/blob/87768755e8653e4be5f29945b56fd05a5246d5a8/core/lib/dal/src/blocks_dal.rs#L1969)). This would allow to check the query in compilation time. In this particular case, there's a much simpler solution though: just use 0 as the default value if the batch number isn't supplied.

Reply:
[Fixed](https://github.com/matter-labs/zksync-era/pull/2674/commits/63e7ada3b727ad1f61cbb70155a3e6972dbf7901). ✅